test/test_testing.py

```diff
@@ -1055,68 +1016,79 @@ def test_mapping_mismatching_values_msg(self, device):
             with self.assertRaisesRegex(AssertionError, r"key\s+'b'"):
                 fn(actual, expected)
 
-    @onlyCPU
-    def test_type_inequality(self, device):
-        actual = torch.empty(2, device=device)
+    def test_type_inequality(self):
+        actual = torch.empty(2)
         expected = actual.tolist()
 
         for fn in self.assert_fns_with_inputs(actual, expected):
             with self.assertRaisesRegexs(UsageError, str(type(actual)), str(type(expected))):
                 fn()
 
-    @onlyCPU
-    def test_unknown_type(self, device):
+    def test_unknown_type(self):
         actual = "0"
         expected = "0"
 
         for fn in self.assert_fns_with_inputs(actual, expected):
             with self.assertRaisesRegexs(UsageError, str(type(actual))):
                 fn()
 
-    @onlyCPU
-    def test_numpy(self, device):
-        tensor = torch.rand(2, 2, dtype=torch.float32, device=device)
+    def test_numpy(self):
+        tensor = torch.rand(2, 2, dtype=torch.float32)
         actual = tensor.numpy()
         expected = actual.copy()
 
         for fn in self.assert_fns_with_inputs(actual, expected):
             fn()
 
-    @onlyCPU
-    def test_scalar(self, device):
-        tensor = torch.rand(1, device=device)
+    def test_scalar(self):
+        tensor = torch.rand(1)
         actual = expected = tensor.item()
 
         for fn in self.assert_fns_with_inputs(actual, expected):
             fn()
 
-    @onlyCPU
-    def test_msg_str(self, device):
+    def test_msg_str(self):
         msg = "Custom error message!"
 
-        actual = torch.tensor(1, device=device)
-        expected = torch.tensor(2, device=device)
+        actual = torch.tensor(1)
+        expected = torch.tensor(2)
 
         for fn in self.assert_fns_with_inputs(actual, expected):
             with self.assertRaisesRegex(AssertionError, msg):
                 fn(msg=msg)
 
-    @onlyCPU
-    def test_msg_callable(self, device):
+    def test_msg_callable(self):
         msg = "Custom error message!"
 
         def make_msg(actual, expected, trace):
             return msg
 
-        actual = torch.tensor(1, device=device)
-        expected = torch.tensor(2, device=device)
+        actual = torch.tensor(1)
+        expected = torch.tensor(2)
 
         for fn in self.assert_fns_with_inputs(actual, expected):
             with self.assertRaisesRegex(AssertionError, msg):
                 fn(msg=make_msg)
 
 
-instantiate_device_type_tests(TestAsserts, globals())
+class TestAssertsMultiDevice(TestCase, _TestAssertsMixin):
+    def test_mismatching_device(self, device):
+        actual = torch.empty((), device=device)
+        expected = actual.clone().cpu()
+
+        for fn in self.assert_fns_with_inputs(actual, expected):
+            with self.assertRaisesRegex(AssertionError, "device"):
+                fn()
+
+    def test_mismatching_device_no_check(self, device):
```

Comment:
Add a test with multiple cuda devices (cuda:0 and cuda:1, for example), too. We have a decorator for these tests:

https://github.com/pytorch/pytorch/blob/df8bb5a42b3c2b1c57bbec9a1c60db6aeafe0f20/torch/testing/_internal/common_device_type.py#L732